File: Reqnroll/Chapter16/16.4-ApprovedOutcomePattern/Before/WIMP.Specs/StepDefinitions/AuthenticationStepDefinitions.cs
using Reqnroll;

using WIMP.Specs.Drivers;
using WIMP.Specs.Support;

namespace WIMP.Specs.StepDefinitions;

[Binding]
public class AuthenticationStepDefinitions(AuthenticationApiDriver authApiDriver)
{
    [Given("the restaurant owner is logged in")]
    public async Task GivenTheRestaurantOwnerIsLoggedIn()
    {
        await authApiDriver
            .Login(DomainDefaults.RestaurantOwner, DomainDefaults.Password)
            .Execute();
    }
}

File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/After/src/test/java/com/wimp/specs/stepdefinitions/AuthenticationStepDefinitions.java
package com.wimp.specs.stepdefinitions;

import com.wimp.specs.drivers.AuthenticationApiDriver;
import com.wimp.specs.support.DomainDefaults;
import io.cucumber.java.en.Given;
import org.springframework.beans.factory.annotation.Autowired;

public class AuthenticationStepDefinitions {
    private final AuthenticationApiDriver authenticationApiDriver;

    @Autowired
    public AuthenticationStepDefinitions(AuthenticationApiDriver authenticationApiDriver) {
        this.authenticationApiDriver = authenticationApiDriver;
    }

    @Given("the customer is authenticated")
    public void theCustomerIsAuthenticated() throws Exception {
        authenticationApiDriver.login(DomainDefaults.CUSTOMER_NAME, DomainDefaults.PASSWORD).execute();
    }
}
